src/helpers.ts

```diff
@@ -64,11 +65,63 @@ export async function netrcPath(): Promise<string> {
         `If you're signed up for FlakeHub Cache, make sure that your Actions config has a \`permissions\` block with \`id-token\` set to "write" and \`contents\` set to "read"`,
       );
       actionsCore.debug(`Error while logging into FlakeHub: ${e}`);
+
+      await annotateWorkflowJobMissingPermissions();
     }
     return destinedNetrcPath;
   }
 }
 
+async function annotateWorkflowJobMissingPermissions(): Promise<void> {
+  try {
+    const workflowFileName = process.env["GITHUB_WORKFLOW"];
+    if (!workflowFileName) {
+      return;
+    }
+
+    const jobName = process.env["GITHUB_JOB"];
+    if (!jobName) {
+      return;
+    }
+
+    let matchedLines = 0;
+    let lineNumber = undefined;
+    const lines = (
+      await fs.readFile(workflowFileName, { encoding: "utf8" })
```

Comment:
if the workflow has a name, GITHUB_WORKFLOW will be that name, and there is no good way to get the actual file at that point. We could get clever and search around for the name ... or search for the jobname and/or workflow name in files to find a match ... or raise a general warning.

Reply:
(dropped from this pr)